src/coreclr/jit/morph.cpp

```diff
@@ -323,6 +323,222 @@ GenTree* Compiler::fgMorphExpandCast(GenTreeCast* tree)
             }
         }
     }
+
+    // This if check needs to be changed to make sure we only
+    // block casts which are already Fixed UP.
+    do
+    {
+        if (!tree->gtOverflow() && varTypeIsFloating(srcType) && varTypeIsIntegral(dstType) && !varTypeIsSmall(dstType))
+        {
+            if ((dstType == TYP_LONG) && (srcType == TYP_FLOAT))
```

Comment:
A lot of this looks to be duplicated from `gtNewSimdCvtNode`, can it not just call that?

Reply:
Then we will be using ```vcvttps2udq/vcvttps2dq/vcvttpd2uqq/vcvttpd2qq``` instead of vcvttss2usi/vcvttss2si/vcvttsd2usi/vcvttsd2si```. Should that be okay?

Reply:
Ah, I see. It's essentially all the same code, the only real difference is whether its using the scalar forms of the instructions.

Perhaps that can be controlled by a parameter or switch, so that the bulk of the complex logic can be shared?

Reply:
Hi Tanner,
I can look at this once we move past the issue with morph. Also, in general, wouldn't it complicate the code in ```gtNewSimdCvtNode```?